pkg/operator/controller/ingress/deployment.go

```diff
@@ -567,13 +573,36 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		})
 	}
 
+	enableDCM := false
+	if ingressControllerDCMEnabled {
+		enableDCM = true
+	}
 	dynamicConfigOverride := unsupportedConfigOverrides.DynamicConfigManager
-	if v, err := strconv.ParseBool(dynamicConfigOverride); err == nil && v {
+	if v, err := strconv.ParseBool(dynamicConfigOverride); err == nil {
+		// Config override can still be used to opt out from DCM.
+		enableDCM = v
+	}
+
+	if enableDCM {
 		env = append(env, corev1.EnvVar{
 			Name:  RouterHAProxyConfigManager,
 			Value: "true",
+		}, corev1.EnvVar{
+			Name:  RouterHAProxyBlueprintRoutePoolSize,
+			Value: "0",
+		})
+
+		// Max dynamic servers override is only taken into account when DCM is enabled.
+		routerHAProxyMaxDynamicServersValue := RouterHAProxyMaxDynamicServersDefaultValue
+		if v, err := strconv.Atoi(unsupportedConfigOverrides.MaxDynamicServers); err == nil && v > 0 {
```

Comment:
It looks like this allows users to set it to a value > 1, which we don't yet support.  Should there be a validation to make sure it doesn't exceed 1?

Reply:
`1` offers the maximum benefit with minimal downside. However, there may be some "brave customers" who’d like to go beyond this limit. I wanted to provides some middle ground between the recommended default and experimental usage.

Reply:
It turns out we allow value > 1 in the unsupportedConfigOverride.

Reply:
Regarding converting string to int here, see https://github.com/openshift/cluster-ingress-operator/pull/1159/files#r1852667015